File: backend/internal/domain/entities/module.go
package entities

import (
	"time"

	"github.com/google/uuid"
)

type Module struct {
	ID        uuid.UUID
	CourseID  uuid.UUID
	OrgID     uuid.UUID
	Title     string
	Position  int
	CreatedAt time.Time
	UpdatedAt time.Time
}

File: frontend/src/types/module.ts
export interface ModuleDTO {
  id: string
  course_id: string
  title: string
  order: number
  created_at: string
  updated_at: string
}

export interface CreateModuleInput {
  title: string
  order?: number
}

export interface UpdateModuleInput {
  title?: string
  order?: number
}
